resources/providers/cluster_provider.go

```diff
@@ -0,0 +1,58 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package providers
+
+import (
+	"context"
+	"fmt"
+	"github.com/elastic/cloudbeat/config"
+	"github.com/elastic/cloudbeat/resources/providers/awslib"
+	k8s "k8s.io/client-go/kubernetes"
+)
+
+type ClusterNameProviderAPI interface {
+	GetClusterName(ctx context.Context, cfg config.Config) (string, error)
+}
+
+type ClusterNameProvider struct {
+	KubernetesClusterNameProvider KubernetesClusterNameProviderApi
+	EKSMetadataProvider           awslib.MetadataProvider
+	EKSClusterNameProvider        awslib.ClusterNameProvider
+	KubeClient                    k8s.Interface
```

Comment:
The reason I decided to not put them in the same interface is that I didn't want to create a dependency between the interface.

The Kubernetes API requires this interface -> `GetClusterName(cfg config.Config, client k8s.Interface) (string, error)`
while the EKS one requires `GetClusterName(ctx context.Context, cfg aws.Config, instanceId string) (string, error)`

I guess that it is possible with some work to make it work so it will use the same interface but then:
1. It will create dependencies between the different interfaces - do we really wish to do it?  same signature. The only reason I created the interfaces is for mock/test purposes.
2. It will enforce the EKS cluster_name_provider to use the EC2 provider. I was a bit against it since I think that the providers shouldn't use each other, I am a bit worried it can cause a circular dependency at some point. I see the providers as agnostic to each other.

That being said, if you think otherwise I'd like to hear your thoughts on it 👍